### PR TITLE
discourse: Updates and fixes

### PR DESCRIPTION
**Pull request by talyz**
###### Motivation for this change
Add the stuff that was missing in #127931, completing the plugin system update:

- Add an `update-plugins` action to the updater
- Update the documentation to reflect the plugin system changes
- Upgrade plugin versions and add missing metadata
- Provide a package with all plugins enabled, useful to avoid local builds when all plugins are wanted

Also:
- Make the updater remove the architecture bound platforms from `Gemfile.lock` and `libv8` architecture bound entries - without this, discourse fails to start with the latest bundler version
- Update Discourse to version `2.7.5`
- Use Terser when building the assets

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ryantm — APPROVED** [review]
Looks good

File: pkgs/servers/web-apps/discourse/plugins/discourse-canned-replies/default.nix [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
     repo = "discourse-canned-replies";
-    rev = "7ee748f18a276aca42185e2079c1d4cadeecdaf8";
-    sha256 = "0j10kxfr6v2rdd58smg2i7iac46z74qnnjk8b91jd1svazhis1ph";
+    rev = "e67bab68594878f50b8fd8ca866f4c168fb37ea4";
+    sha256 = "1gi1ivvqwijmcyyn3hmhfyz4slckm4dl7spkw4q7vyqx7gxmg5vn";
+  };
+  meta = with lib; {
+    homepage = "https://github.com/discourse/discourse-canned-replies";
+    maintainers = with maintainers; [ talyz ];
+    license = licenses.gpl2;
```
Comment (lukegb) [inline pkgs/servers/web-apps/discourse/plugins/discourse-canned-replies/default.nix:14]: gpl2Only: https://github.com/discourse/discourse-canned-replies/blob/main/package.json#L3

**Review by lukegb — APPROVED** [review]
LGTM, one nit

File: pkgs/servers/web-apps/discourse/plugins/discourse-canned-replies/default.nix (hunk 1 above)
Comment (talyz) [reply pkgs/servers/web-apps/discourse/plugins/discourse-canned-replies/default.nix:14]: Oh, right! Thanks!

File: pkgs/servers/web-apps/discourse/update.py [hunk 2]
```diff
@@ -35,6 +38,15 @@ def tags(self) -> Iterable[str]:
         versions.sort(key=lambda x: LooseVersion(x.replace('v', '')), reverse=True)
         return versions
 
+    @property
+    def latest_commit_sha(self) -> str:
+        if self._latest_commit_sha is None:
+            r = requests.get(f'https://api.github.com/repos/{self.owner}/{self.repo}/commits?per_page=1')
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/discourse/update.py:44]: Does this use a token?

Comment (talyz) [reply pkgs/servers/web-apps/discourse/update.py:44]: No, you can make these API requests without one. Having one probably lifts the rate limit, but it's at 60 requests without a token, which is plenty.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129464` at 39dd141e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/f86899ec9254e806195883d8c9d28df5">discourseAllPlugins</a></li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

bundler is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/servers/web-apps/discourse/default.nix:217:5:
```
    |
217 |     buildInputs = [
    |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `replace`.
Near pkgs/servers/web-apps/discourse/default.nix:4:9:
```
  |
4 | , ruby, replace, gzip, gnutar, git, cacert, util-linux, gawk
  |         ^
```
</li>
    <li>warning: unused-argument

Unused argument: `meta`.
Near pkgs/servers/web-apps/discourse/default.nix:56:7:
```
   |
56 |     , meta ? null
   |       ^
```
</li>
    <li>warning: unused-argument

Unused argument: `src`.
Near pkgs/servers/web-apps/discourse/default.nix:58:7:
```
   |
58 |     , src
   |       ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by talyz** [comment]
Rebased on `master` and pushed another update which removes the architecture bound `libv8` `Gemfile.lock` entries, since they caused runtime errors with the latest version of bundler. The update also makes sure these architecture bound entries are always removed from the lock file by removing the platforms in the updater script.

**Comment by talyz** [comment]
I previously forgot about handling the plugin ruby dependencies in the updater, but that's sorted now. I also expanded the manual section to include instructions on how to package plugins with ruby deps.

File: pkgs/servers/web-apps/discourse/update.py [hunk 3]
```diff
@@ -8,17 +8,22 @@
 import re
 import logging
 import subprocess
-import pathlib
```
Comment (dpausp) [inline pkgs/servers/web-apps/discourse/update.py:11]: This breaks `update`, line 169 also needs `pathlib.Path` -> `Path`

**Review by dpausp — CHANGES_REQUESTED** [review]
`update-plugins` works as expected, there are some new plugin versions. `update` is broken.

File: pkgs/servers/web-apps/discourse/update.py (hunk 3 above)
Comment (talyz) [reply pkgs/servers/web-apps/discourse/update.py:11]: Oh, whoops. Fixed it and added the latest update to `2.7.5`.

**Review by dpausp — APPROVED** [review]
Looks good now. 

Tested all `update.py` commands and also also played around a bit with Discourse itself: tested posting, looked at the admin dashboard and plugin list, and ran a backup (on 21.05 without the terser commit).

**Comment by dpausp** [comment]
As a side note, if someone wants to try the beta version (tests-passed), `update.py update v2.8.0.beta2` works and Discourse is running properly.

**Pull request merged**